pkg/konfluxgen/konfluxgen.go

```diff
@@ -428,6 +447,149 @@ func Generate(cfg Config) error {
+	if err != nil {
+		return fmt.Errorf("failed to parse application template: %w", err)
+	}
+
+	dockerfileComponentTemplate, err := template.New("dockerfile-component.template.yaml").Funcs(funcs).ParseFS(DockerfileComponentTemplate, "*.yaml")
+	if err != nil {
+		return fmt.Errorf("failed to parse dockerfile component template: %w", err)
+	}
+
+	imageRepositoryTemplate, err := template.New("imagerepository.template.yaml").Funcs(funcs).ParseFS(ImageRepositoryTemplate, "*.yaml")
+	if err != nil {
+		return fmt.Errorf("failed to parse dockerfile component template: %w", err)
+	}
+
+	pipelineRunTemplate, err := template.New("pipeline-run.template.yaml").Delims("{{{", "}}}").Funcs(funcs).ParseFS(PipelineRunTemplate, "*.yaml")
+	if err != nil {
+		return fmt.Errorf("failed to parse pipeline run push template: %w", err)
+	}
+
+	pipelineFBCBuildTemplate, err := template.New("fbc-builder.yaml").Delims("{{{", "}}}").Funcs(funcs).ParseFS(PipelineFBCBuildTemplate, "*.yaml")
+	if err != nil {
+		return fmt.Errorf("failed to parse pipeline run push template: %w", err)
+	}
+
+	appKey := Truncate(Sanitize(cfg.ApplicationName))
+	componentKey := Truncate(Sanitize(fmt.Sprintf("%s-index", cfg.ApplicationName)))
+
+	if err := os.RemoveAll(filepath.Join(cfg.ResourcesOutputPath, ApplicationsDirectoryName, appKey)); err != nil {
+		return fmt.Errorf("failed to cleanup application directory %q: %w", appKey, err)
+	}
+
+	appPath := filepath.Join(cfg.ResourcesOutputPath, ApplicationsDirectoryName, appKey, fmt.Sprintf("%s.yaml", appKey))
+	if err := os.MkdirAll(filepath.Dir(appPath), 0777); err != nil {
+		return fmt.Errorf("failed to create directory for %q: %w", appPath, err)
+	}
+
+	config := DockerfileApplicationConfig{
+		ApplicationName: cfg.ApplicationName,
+		ComponentName:   componentKey,
+		ReleaseBuildConfiguration: cioperatorapi.ReleaseBuildConfiguration{
+			Metadata: cfg.Metadata,
+		},
+		//Path:                      c.Path,
+		ProjectDirectoryImageBuildStepConfiguration: cioperatorapi.ProjectDirectoryImageBuildStepConfiguration{
+			To: cioperatorapi.PipelineImageStreamTagReference(componentKey),
+			ProjectDirectoryImageBuildInputs: cioperatorapi.ProjectDirectoryImageBuildInputs{
+				DockerfilePath: cfg.DockerfilePath,
+				ContextDir:     cfg.ContextDirPath,
+			},
+		},
+		Pipeline:                      FBCBuild,
+		Tags:                          append(cfg.Tags, "latest"),
+		BuildArgs:                     cfg.BuildArgs,
+		Hermetic:                      fmt.Sprintf("%t", cfg.IsHermetic),
+		AdditionalTektonCELExpression: cfg.AdditionalTektonCELExpressions,
+	}
+
+	// Generate application yaml
+	buf := &bytes.Buffer{}
+	if err := applicationTemplate.Execute(buf, config); err != nil {
+		return fmt.Errorf("failed to execute template for application %q: %w", appKey, err)
+	}
+	if err := os.WriteFile(appPath, buf.Bytes(), 0777); err != nil {
+		return fmt.Errorf("failed to write application file %q: %w", appPath, err)
+	}
```

Comment:
Duplications from [konfluxgen.Generate()](https://github.com/openshift-knative/hack/blob/f2a9e86ff33b53185aabec1a55f84660f3523a15/pkg/konfluxgen/konfluxgen.go#L326). But before refactoring it, I want to align on the general approach first

Reply:
Agree there is a lot of duplication

Reply:
we could "augment" `collectConfigurations` with an optional function for the additional images/components

Reply:
:+1: 
Updated